### ISSUE TITLE
Discrepancies with multiple addresses in a wallet

### ISSUE THREAD

**Issue by jseagrave21** [CityOfZion/neo-python#605]
### Current behavior
Using `account` to display an address's balance hides other balances within the wallet and makes them inaccessible. Here is what I see:
```
Wallet {
    "path": "neo-privnet.wallet",
    "addresses": [
        {
            "version": 0,
            "script_hash": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "99999900.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "74674.9999"
            },
            "is_watch_only": false
        },
        {
            "version": 0,
            "script_hash": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "100.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "4.995"
            },
            "is_watch_only": false
        },
        {
            "address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
            "script_hash": "10d951cc11ce0d3b562acd0c0bbb183707fff64c",
            "tokens": null
        }
    ],
    "height": 9572,
    "percent_synced": 100,
    "synced_balances": [
        "[NEO]: 99999900.0 ",
        "[NEOGas]: 74629.9999 "
    ],
    "public_keys": [
        {
            "Address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "Public Key": "031a6c6fbbdf02ca351745fa86b9ba5a9452d785ac4f7fc2b7548ca2a46c4fcf4a"
        },
        {
            "Address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR",
            "Public Key": "0322bea896b923fb37a80dcb79d22061af85751b7140878987110b988e9a3e353f"
        },
        {
            "Address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
            "Public Key": "031b790f7589069981e04e96fb072f5433bd7f5acc82d46a04a0040d4e8ee36834"
        }
    ],
    "tokens": [],
    "claims": {
        "available": "374.0",
        "unavailable": "999.999"
    }
}
neo> account AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps
{
    "version": 0,
    "script_hash": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
    "frozen": false,
    "votes": [],
    "balances": {
        "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "50.0",
        "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "5.0"
    }
}


neo> wallet
[I 180912 23:31:54 UserWallet:538] Script hash b'#\xba\'\x03\xc52c\xe8\xd6\xe5"\xdc2 39\xdc\xd8\xee\xe9' <class 'bytes'>
[I 180912 23:31:54 UserWallet:538] Script hash b'\xbf\xdc\x06,\x82\xde`S\x94C\xd2\x13\xa9\xae\xd8\x97q\xd1*\xe8' <class 'bytes'>
[I 180912 23:31:54 UserWallet:538] Script hash b'\x10\xd9Q\xcc\x11\xce\r;V*\xcd\x0c\x0b\xbb\x187\x07\xff\xf6L' <class 'bytes'>
Wallet {
    "path": "neo-privnet.wallet",
    "addresses": [
        {
            "version": 0,
            "script_hash": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "99999950.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "74674.9899"
            },
            "is_watch_only": false
        },
        {
            "address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR",
            "script_hash": "bfdc062c82de60539443d213a9aed89771d12ae8",
            "tokens": null
        },
        {
            "version": 0,
            "script_hash": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "50.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "5.0"
            },
            "is_watch_only": false
        }
    ],
    "height": 9573,
    "percent_synced": 100,
    "synced_balances": [
        "[NEO]: 100000000.0 ",
        "[NEOGas]: 74679.9899 "
    ],
    "public_keys": [
        {
            "Address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "Public Key": "031a6c6fbbdf02ca351745fa86b9ba5a9452d785ac4f7fc2b7548ca2a46c4fcf4a"
        },
        {
            "Address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR",
            "Public Key": "0322bea896b923fb37a80dcb79d22061af85751b7140878987110b988e9a3e353f"
        },
        {
            "Address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
            "Public Key": "031b790f7589069981e04e96fb072f5433bd7f5acc82d46a04a0040d4e8ee36834"
        }
    ],
    "tokens": [],
    "claims": {
        "available": "374.000552",
        "unavailable": "1007.999"
    }
}
neo> account AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR
Account AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR not found
```

### Expected behavior
Balances of every address within a wallet should be displayed and using `account` to display an account state of a certain address should not affect other addresses within the wallet.

### How to reproduce

- Access the privatenet using np-prompt -p (or testnet if you'd like, I was using privatenet)
- Open a wallet and use `create_addr` to create 2 (or more) addresses within the wallet
- Send assets to both addresses
- Using `wallet`, check to see if you can see all balances of every address within the wallet
- Use `account` to display the balance of the address which has a hidden balance
- Using `wallet`, check to see if you can see all balances of every address within the wallet
- Try to use `account` to display the balance of the wallet that is now hidden and the address cannot be found

### Your environment

Let us know in what environment you're running into the issue:

- OS: Microsoft Windows 10 Pro 10.0.17134 Build 17134
- neo-python version: neo-python v0.7.8-dev
- Python version: 3.7.0

**Comment by jseagrave21** [comment]
Upon further investigation, I found that this issue is actually what appears to be a glitch when running neo-privatenet. In the previous example, `account` did not affect the balance of the address. Instead, somehow the balance completely reset.

I saw a similar issue and asked @belane  about it, and we could resolve it. Below is the example case.
Here are the last three wallet tx shown when using `wallet verbose` in prompt on privatenet.

First tx
```
{
            "txid": "0x84e8b5f8a9c0ceaaacb324046d8047a9dc2fa7477ec9188b6196e6984ac2d05b",
            "size": 557,
            "type": "ContractTransaction",
            "version": 0,
            "attributes": [
                {
                    "usage": 32,
                    "data": "23ba2703c53263e8d6e522dc32203339dcd8eee9"
                }
            ],
            "vout": [
                {
                    "n": 0,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "100",
                    "address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR"
                },
                {
                    "n": 1,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "50",
                    "address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps"
                },
                {
                    "n": 2,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "10",
                    "address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR"
                },
        
        {
                    "n": 3,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "5",
                    "address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps"
                },
                {
                    "n": 4,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "99999850",
                    "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y"
                },
                {
                    "n": 5,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "26058.995",
                    "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y"
                }
            ],
            "vin": [
                {
                    "txid": "0x539084697cc220916cb5b16d2805945ec9f267aa004b6688fbf15e116c846aff",
                    "vout": 0
                },
                {
                    "txid": "0xaa2051096e7be45ed991279a1a4c2678eb886690829a2729f4caa82192ff7f34",
                    "vout": 0
                }
            ],
            "sys_fee": "0",
            "net_fee": "0.005",
            "scripts": [
                {
```        
Second tx
```
{
            "txid": "0xd8c982b92941924fe93ba55b0313dade243ad96b74387b5510e35c419d8719fd",
            "size": 539,
            "type": "ContractTransaction",
            "version": 0,
            "attributes": [
                {
                    "usage": 32,
                    "data": "23ba2703c53263e8d6e522dc32203339dcd8eee9"
                }
            ],
            "vout": [
                {
                    "n": 0,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "25",
                    "address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR"
                },
                {
                    "n": 1,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "2.5",
                    "address": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR"
                },
                {
                    "n": 2,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "25",
                    "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y"
                },
                {
                    "n": 3,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "2.5",
                    "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y"
                }
```
Third tx
```
{
            "txid": "0x054bf85f03b05ff087f8b1255a963d4b7942762ba5ae856ad2e6582be3ee315f",
            "size": 487,
            "type": "ContractTransaction",
            "version": 0,
            "attributes": [
                {
                    "usage": 32,
                    "data": "23ba2703c53263e8d6e522dc32203339dcd8eee9"
                }
            ],
            "vout": [
                {
                    "n": 0,
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "50",
                    "address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps"
                },
                {
                    "n": 1,
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "5",
                    "address": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps"
                }
            ]
```
When using the `wallet` command, the wrong balance is displayed for AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR
```
Wallet {
    "path": "neo-privnet.wallet",
    "addresses": [
        {
            "version": 0,
            "script_hash": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "99999850.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "74664.9949"
            },
            "is_watch_only": false
        },
        {
            "version": 0,
            "script_hash": "AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "100.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "10.0"
            },
            "is_watch_only": false
        },
        {
            "version": 0,
            "script_hash": "AHJxoC87MMGJHw353SMzZQLLuhR5QZ5Jps",
            "frozen": false,
            "votes": [],
            "balances": {
                "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b": "50.0",
                "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7": "5.0"
            },
            "is_watch_only": false
        }
    ]
```
You can see that address AZGLHgV4sBsAQLYioetrJpTeSFNLsGzJdR is completely missing the 2nd tx and its balance should be 125 'neo' and 12.5 'gas'.

`wallet rebuild` does not solve the problem and there were several blocks between the first, second, and third txs. Also, all txs appear on the blockchain when queried by RPC. (these were all questions asked by @belane )

So, now I am thinking this issue may be related to #578 . Any thoughts @ixje @localhuman ?

**Comment by ixje** [comment]
is this still present in the latest dev branch?

**Comment by jseagrave21** [comment]
I will check.

**Comment by jseagrave21** [comment]
@ixje I just verified on the latest branch on privatenet that this behavior is still exhibited. I did exactly the same steps as previously mentioned [here](https://github.com/CityOfZion/neo-python/issues/605#issuecomment-421662084)

**Comment by jseagrave21** [comment]
@ixje Okay, I will try to re-create it again. Only on the `refactor-prompt` branch now.

**Comment by jseagrave21** [comment]
While trying to re-create this issue, I have found that I am unable to use the `sendmany` function as before. I receive no errors, so I am unsure what is happening. Here is what I did. Please note the change in synced balances. If I try to send another transaction, I receive an "Insufficient Funds" error since I have no more available `vouts`. Here is what I did:
```
neo> wallet address create 2
Created 2 new addresses: ['AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV', 'AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm']

neo> wallet
[I 190105 13:37:50 UserWallet:480] Script hash b'#\xba\'\x03\xc52c\xe8\xd6\xe5"\xdc2 39\xdc\xd8\xee\xe9' <class 'bytes'>
[I 190105 13:37:50 UserWallet:480] Script hash b'\xd93\xc6@\xa2L\xf3JL\xc9V\xfa\xd4\xd0\xc9\xe1\x95@\x84\xe0' <class 'bytes'>
[I 190105 13:37:50 UserWallet:480] Script hash b'\xa6yRNsR(;\x81:p\x86#Q\x1b\xa0\x13\xc2?D' <class 'bytes'>
Wallet {
    "path": "neo-privnet.wallet",
    "addresses": [
        {
            "version": 0,
            "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "script_hash": "e9eed8dc39332032dc22e5d6e86332c50327ba23",
            "frozen": false,
            "votes": [],
            "balances": [
                {
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "100000000.0"
                },
                {
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "16024.0"
                }
            ],
            "is_watch_only": false
        },
        {
            "address": "AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV",
            "script_hash": "d933c640a24cf34a4cc956fad4d0c9e1954084e0",
            "tokens": null
        },
        {
            "address": "AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm",
            "script_hash": "a679524e7352283b813a708623511ba013c23f44",
            "tokens": null
        }
    ],
    "height": 2013,
    "percent_synced": 100,
    "synced_balances": [
        "[NEO]: 100000000.0 ",
        "[NEOGas]: 16024.0 "
    ],
    "public_keys": [
        {
            "Address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "Public Key": "031a6c6fbbdf02ca351745fa86b9ba5a9452d785ac4f7fc2b7548ca2a46c4fcf4a"
        },
        {
            "Address": "AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV",
            "Public Key": "02c5f2741862553388dbaf40720cc11aff39b17a8abe5751f6e2e0fae221c5f320"
        },
        {
            "Address": "AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm",
            "Public Key": "03bc11c2769c736ade264845273567a695d82dee040588d70c77bb4c7e46359b8b"
        }
    ],
    "tokens": [],
    "claims": {
        "available": "0.0",
        "unavailable": "48.0"
    }
}

neo> wallet sendmany 4
Outgoing Number 1
Asset to send: neo
Address to: AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV
Amount to send: 500
Outgoing Number 2
Asset to send: gas
Address to: AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV
Amount to send: 100
Outgoing Number 3
Asset to send: neo
Address to: AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm
Amount to send: 250
Outgoing Number 4
Asset to send: gas
Address to: AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm
Amount to send: 50
sending with fee: 0.0
[Password]> ***
[I 190105 13:40:42 Transaction:619] Verifying transaction: b'bc2530f7705fc011ac28e234e548c4d01cd47b58aaac557604bd26fe4b763b2e'
Relayed Tx: bc2530f7705fc011ac28e234e548c4d01cd47b58aaac557604bd26fe4b763b2e

neo> show tx bc2530f7705fc011ac28e234e548c4d01cd47b58aaac557604bd26fe4b763b2e
Could not find transaction for hash bc2530f7705fc011ac28e234e548c4d01cd47b58aaac557604bd26fe4b763b2e

neo> wallet
[I 190105 13:41:33 UserWallet:480] Script hash b'#\xba\'\x03\xc52c\xe8\xd6\xe5"\xdc2 39\xdc\xd8\xee\xe9' <class 'bytes'>
[I 190105 13:41:33 UserWallet:480] Script hash b'\xd93\xc6@\xa2L\xf3JL\xc9V\xfa\xd4\xd0\xc9\xe1\x95@\x84\xe0' <class 'bytes'>
[I 190105 13:41:33 UserWallet:480] Script hash b'\xa6yRNsR(;\x81:p\x86#Q\x1b\xa0\x13\xc2?D' <class 'bytes'>
Wallet {
    "path": "neo-privnet.wallet",
    "addresses": [
        {
            "version": 0,
            "address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "script_hash": "e9eed8dc39332032dc22e5d6e86332c50327ba23",
            "frozen": false,
            "votes": [],
            "balances": [
                {
                    "asset": "0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b",
                    "value": "100000000.0"
                },
                {
                    "asset": "0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7",
                    "value": "16024.0"
                }
            ],
            "is_watch_only": false
        },
        {
            "address": "AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV",
            "script_hash": "d933c640a24cf34a4cc956fad4d0c9e1954084e0",
            "tokens": null
        },
        {
            "address": "AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm",
            "script_hash": "a679524e7352283b813a708623511ba013c23f44",
            "tokens": null
        }
    ],
    "height": 2025,
    "percent_synced": 100,
    "synced_balances": [
        "[NEO]: 0.0 ",
        "[NEOGas]: 0.0 "
    ],
    "public_keys": [
        {
            "Address": "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y",
            "Public Key": "031a6c6fbbdf02ca351745fa86b9ba5a9452d785ac4f7fc2b7548ca2a46c4fcf4a"
        },
        {
            "Address": "AbaLJUY2FRfSkuudHAveSKL2htoGp1ugbV",
            "Public Key": "02c5f2741862553388dbaf40720cc11aff39b17a8abe5751f6e2e0fae221c5f320"
        },
        {
            "Address": "AWx79ueFWEd9Nc5jP99yeLRtQgNDaAPuRm",
            "Public Key": "03bc11c2769c736ade264845273567a695d82dee040588d70c77bb4c7e46359b8b"
        }
    ],
    "tokens": [],
    "claims": {
        "available": "0.0",
        "unavailable": "0.0"
    }
}
```

**Comment by jseagrave21** [comment]
@ixje on a fresh pull of the `refactor-prompt` branch, I was unable to reproduce the previously existing behavior. I really appreciated the changes in #796, which I could use to verify my expectations after the confirmation was confirmed. Also, I can verify that the issue solved by #803 was actually resolved. I consider this issue resolved 🎉

**Comment by ixje** [comment]
Thanks for checking all that and reporting on it 💪
